packages/inquirer/examples/regex-validate-input.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Filter and validate progress example
+ */
+
+'use strict';
+const inquirer = require('..');
+
+/* eslint-disable no-promise-executor-return */
+const questions = [
+  {
+    type: 'input',
+    name: 'api_key',
+    message: 'Please enter a valid API key.',
+    validate(input) {
+      return Promise.resolve().then(() => {
+        if (!!input.match(/([a-f0-9]{40})/g)) {
```

Comment:
```suggestion
        if (/([a-f0-9]{40})/g.test(input)) {
```

Getting the type we want directly instead